packages/test/test-service-load/src/smokeTest.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITestDriver } from "@fluid-internal/test-driver-definitions";
+import { IContainer, IFluidCodeDetails } from "@fluidframework/container-definitions/internal";
+import { Loader } from "@fluidframework/container-loader/internal";
+import { assert } from "@fluidframework/core-utils/internal";
+
+import { FileLogger } from "./FileLogger.js";
+import { pkgName, pkgVersion } from "./packageVersion.js";
+import { createCodeLoader } from "./utils.js";
+
+const packageName = `${pkgName}@${pkgVersion}`;
+const codeDetails: IFluidCodeDetails = {
+	package: packageName,
+	config: {},
+};
+
+// The stress smoke test is run before the actual stress test.  Its job is not to actually
+// perform stress, but instead just to confirm that the basic prerequisite capabilities
+// (authentication, container creation, container loading, etc.) are functioning prior to
+// running stress.  If the smoke test isn't working, it will give us a concrete failure
+// to investigate and allow us to skip a (doomed) attempt at running the stress tests.
+
+export async function smokeTest(testDriver: ITestDriver, profileName: string) {
+	const logger = await FileLogger.createLogger({
+		driverType: testDriver.type,
+		driverEndpointName: testDriver.endpointName,
+		profile: profileName,
+		runId: undefined,
+	});
+
+	// Construct the loader
+	const loader = new Loader({
+		urlResolver: testDriver.createUrlResolver(),
+		documentServiceFactory: testDriver.createDocumentServiceFactory(),
+		codeLoader: createCodeLoader(), // For the smoke test, just run with default container runtime options
+		logger,
+	});
+
+	// Verify container creation works
+	const createContainerAndGetUrl = async () => {
+		const createdContainer: IContainer = await loader.createDetachedContainer(codeDetails);
+
+		const testId = Date.now().toString();
+		const request = testDriver.createCreateNewRequest(testId);
+
+		await createdContainer.attach(request);
+		assert(
+			createdContainer.resolvedUrl !== undefined,
+			"Container missing resolved URL after attach",
+		);
+
+		const resolvedUrl = createdContainer.resolvedUrl;
+		createdContainer.dispose();
+
+		return testDriver.createContainerUrl(testId, resolvedUrl);
+	};
+	console.log("Creating container and attaching...");
+	const url = await tryNTimes(createContainerAndGetUrl, 3, 3);
```

Comment:
this seems to be a very different test from the existing stress test, as the existing test only creates a single container, and then has lots of runners party on it. I worry spamming container creates could cause throttling issues that are independent of the kind the test itself would hit. Can't we just do similar validation in the orchestrator on the container that all the runners will use?

Reply:
This also only creates a single container (so now 2 containers per run, one for smoke and one for stress).  tryNTimes is just for retry, with intent that a single flake-out doesn't fail the run.  Shouldn't cause any throttling problems.

Reply:
ahh i see. maybe just reuse runWithRetry from driver utils rather than creating another retry function

Reply:
but i also think you could just do this to the main container the test uses, rather than creating a second container. the main container creation could also use retry

Reply:
I think runWithRetry really only works for errors of type IDriverErrorBase - I'm probably OK limiting retries to just driver (network) errors, but this would retry in fewer scenarios than as currently written.

I favor not reusing the container for two main reasons:
1. I like having the separation between the two - changes to either smoke or stress can be made without worrying about impacting the other
2. The loader options for the stress test's document creation are more complex and randomly selected which could produce inconsistent results across random seeds.  My goal is for smoke to be super-consistent run-to-run such that its failure is a super-strong signal that basic functionality is broken.